controller/api/views.py

```diff
@@ -51,6 +52,40 @@ def passwd(self, request, **kwargs):
         return Response({'status': 'password set'})
 
 
+class TokenManagementViewSet(GenericViewSet,
+                             mixins.DestroyModelMixin):
+    serializer_class = serializers.UserSerializer
+    permission_classes = [permissions.CanRegenerateToken]
+
+    def get_queryset(self):
+        return User.objects.filter(pk=self.request.user.pk)
+
+    def get_object(self):
+        return self.get_queryset()[0]
+
+    def regenerate(self, request, **kwargs):
+        obj = self.get_object()
+
+        if 'all' in request.data:
+            for user in User.objects.exclude(username='AnonymousUser'):
```

Comment:
rather than checking for the username, we can check `.is_anonymous()` instead to achieve the same effect. That may be more portable.